File: userStatuses/python/userStatuses.py
# Copyright 2018 Google Inc. All rights reserved.

# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at

#     http://www.apache.org/licenses/LICENSE-2.0

# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

"""
A python example of calling the RCS deactivation API.
"""

from httplib2 import Http
from oauth2client.service_account import ServiceAccountCredentials

import config


def main():
    print('Looking up account ' + config.MSISDN)

    scopes = ['https://www.googleapis.com/auth/mobilenetworkservices']
    credentials = ServiceAccountCredentials.from_json_keyfile_name(
        'resources/service-account-credentials.json', scopes=scopes)
    http_auth = credentials.authorize(Http())

    serviceUrl = 'https://mobilenetworkservices.googleapis.com/v2/plmns/' + config.MCCMNC + '/userStatuses?serviceName=RCS&filter=msisdn=' + config.MSISDN

    print("User statuses URL: " + serviceUrl);

    http_auth = credentials.authorize(Http())

    resp, content = http_auth.request(serviceUrl, method='GET')

    print('Received response:');
    print(resp);
    print(content);

if __name__ == '__main__':
    main()

File: userStatuses/nodejs/src/userStatuses.js
// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the License);
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an AS IS BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

'use strict';

const {google} = require('googleapis');
const googleapis_common_1 = require('googleapis-common');

const serviceAccountJsonObject = require('../service-account-credentials.json');

const config = require('./config');


class UserStatuses {
	constructor(mccmnc) {
		this.mccmnc = mccmnc;
		this.context = {
			_options: {},
			google,
		};

		this.authClient = new google.auth.JWT(
			serviceAccountJsonObject.client_email,
			null,
			serviceAccountJsonObject.private_key,
			['https://www.googleapis.com/auth/mobilenetworkservices']
		);
	}

	userStatus(msisdn) {
		const serviceUrl = `https://mobilenetworkservices.googleapis.com/v2/plmns/${this.mccmnc}/userStatuses?serviceName=RCS&filter=msisdn=${msisdn}`;

		const parameters = {
			options: {
				url: serviceUrl
			},
			params: {
				auth: this.authClient
			},
			requiredParams: [],
			pathParams: [],
			context: this.context,
		};
		
		googleapis_common_1.createAPIRequest(parameters).then((res) => {
			console.log(res);
			console.log(res.data);
		});
	}
}

new UserStatuses(config.mccmnc).userStatus(config.msisdn);
